### DISCUSSION TITLE
Web Components support with Lit Elements

### DISCUSSION THREAD

**Discussion by itaymendel** [teambit/bit#4823] [Announcements]
Web Components allows us create reuse-able custom elements that encapsulate their functionality from the consuming application.  
[Lit](https://lit.dev) is a framework for building custom elements on top of the Web Components standards.

If you want to build and share Web Components with Bit, you can now use this [Lit aspect](https://bit.dev/teambit/web-components/lit) in your Bit workspace.  
The Lit Component Development Environment features:

* Workspace template to get started quickly.
* Component generator to simplify creation of components.
* Native Jest support for testing.
* Native TypeScript support.
* Rendering with HTML.
* Component documentation driven by MD or MDX with live playground.

Additional resources:

* [Lit aspect shared on Bit.dev](https://bit.dev/teambit/web-components/lit)
* [Code repository on GitHub](https://github.com/teambit/bit-lit)

## Getting started

> Make sure you have Bit [installed on your computer](https://harmony-docs.bit.dev/getting-started/installing-bit).

To create a new workspace from a template run the following command:

```sh
bit new lit my-lit-workspace --aspect teambit.web-components/lit
```

`cd` to project directory and open `workspace.jsonc` to configure the workspace to your liking.

Create your first component:

```sh
bit create lit my-component
```

Ensure all dependencies installed:

```sh
bit install
```

Start the local dev server:

```sh
bit start
```

From here you can `tag` and `export` Web Components and use them in any web application!

**Comment by rapitkan** [comment]
Lit workspace is just what I've been looking for and it almost works fine. There are only two errors on the project template.

<img width="911" alt="image" src="https://github.com/teambit/bit/assets/2033731/23374ce8-f5df-43b7-bed6-ecb996d0585f">

<img width="880" alt="image" src="https://github.com/teambit/bit/assets/2033731/0c3de484-e6e2-48f5-bf3a-fcb285fd00ba">

Would you @itaymendel have the knowledge of solving those issues? How to please typescript?
